criu/image.c

```diff
@@ -110,8 +131,87 @@ int check_img_inventory(bool restore)
 	return ret;
 }
 
+/**
+ * Check if the 'plugins' field in the inventory image contains
+ * the specified plugin name. If found, the plugin is removed
+ * from the linked list.
+ */
+bool check_and_remove_inventory_plugin(const char *name, size_t n)
+{
+	if (n_inventory_plugins == -1)
+		return true; /* backwards compatibility */
+
+	if (n_inventory_plugins > 0) {
+		struct inventory_plugin *p, *tmp;
+		list_for_each_entry_safe(p, tmp, &inventory_plugins_list, node) {
+			if (!strncmp(name, p->name, n)) {
+				xfree(p->name);
+				list_del(&p->node);
```

Comment:
xfree(p)?

Reply:
Thank you for catching this! I've updated the pull request.